data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/remedial_action/range_action/PstWithRange.java

```diff
@@ -0,0 +1,225 @@
+        double minValue = range.getMin();
+        return convertTapToAngle(network, Math.max(lowTapPosition, (int) getExtremumValueWithRange(range, getCurrentTapPosition(network), minValue)));
+    }
+
+    @Override
+    public double getMaxValueWithRange(Network network, Range range) {
+        double maxValue = range.getMax();
+        return convertTapToAngle(network, Math.min(highTapPosition, (int) getExtremumValueWithRange(range, getCurrentTapPosition(network), maxValue)));
+    }
+
+    @Override
+    public double getMaxNegativeVariation(Network network) {
+        return Math.abs(convertTapToAngle(network, (int) getMinValue(network)) - convertTapToAngle(network, getCurrentTapPosition(network)));
+    }
+
+    @Override
+    public double getMaxPositiveVariation(Network network) {
+        return Math.abs(convertTapToAngle(network, (int) getMaxValue(network))  - convertTapToAngle(network, getCurrentTapPosition(network)));
+    }
+
+    private int getCurrentTapPosition(Network network) {
+        return network.getTwoWindingsTransformer(networkElement.getId()).getPhaseTapChanger().getTapPosition();
+    }
+
+    private double convertTapToAngle(Network network, int tap) {
+        TwoWindingsTransformer transformer = network.getTwoWindingsTransformer(networkElement.getId());
+        return transformer.getPhaseTapChanger().getStep(tap).getAlpha();
+    }
+
+    private double getExtremumValueWithRange(Range range, double currentTapPosition, double extremumValue) {
+        switch (range.getRangeType()) {
+            case ABSOLUTE_FIXED:
+                switch (range.getRangeDefinition()) {
+                    case STARTS_AT_ONE:
+                        return lowTapPosition + extremumValue - 1;
+                    case CENTERED_ON_ZERO:
+                        return ((double) lowTapPosition + highTapPosition) / 2 + extremumValue;
+                    default:
+                        throw new FaraoException("Unknown range definition");
+                }
+            case RELATIVE_FIXED:
+                return initialTapPosition + extremumValue;
+            case RELATIVE_DYNAMIC:
+                return currentTapPosition + extremumValue;
+            default:
+                throw new FaraoException("Unknown range type");
+        }
+    }
+
+    /**
+     * Change tap position of the PST pointed by the network element.
+     *
+     * @param network: network to modify
+     * @param finalAngle: angle value to set on the PST.
+     */
+    @Override
+    public void apply(Network network, double finalAngle) {
+
+        PhaseTapChanger phaseTapChanger = checkValidPstAndGetPhaseTapChanger(network);
+        int setpoint = computeTapPosition(finalAngle, phaseTapChanger);
+        if (phaseTapChanger.getHighTapPosition() - phaseTapChanger.getLowTapPosition() + 1 >= setpoint && setpoint >= 1) {
+            phaseTapChanger.setTapPosition(setpoint + phaseTapChanger.getLowTapPosition() - 1);
```

Comment:
computeTapPosition returns a tap with the same conventions as the network, not with the "starts at one" convention